libs/compute/tests/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include(HPX_Message)
+include(HPX_Option)
+
+if (NOT HPX_WITH_TESTS AND HPX_TOP_LEVEL)
+  hpx_set_option(HPX_COMPUTE_WITH_TESTS VALUE OFF FORCE)
+  return()
+endif()
+if (NOT HPX_COMPUTE_WITH_TESTS)
+  hpx_info("    Tests for compute disabled")
+  return()
+endif()
+
+if (HPX_WITH_TESTS_UNIT)
+  add_hpx_pseudo_target(tests.unit.modules.compute)
+  add_hpx_pseudo_dependencies(tests.unit.modules tests.unit.modules.compute)
+  add_subdirectory(unit)
+endif()
+
+if (HPX_WITH_TESTS_REGRESSIONS)
+  add_hpx_pseudo_target(tests.regressions.modules.compute)
+  add_hpx_pseudo_dependencies(tests.regressions.modules tests.regressions.modules.compute)
+  add_subdirectory(regressions)
+endif()
+
+if (HPX_WITH_TESTS_BENCHMARKS)
+  add_hpx_pseudo_target(tests.performance.modules.compute)
+  add_hpx_pseudo_dependencies(tests.performance.modules tests.performance.modules.compute)
+  add_subdirectory(performance)
+endif()
+
+if (HPX_WITH_TESTS_HEADERS)
+  # TODO: Why is this needed?
+  add_library(hpx_includes INTERFACE)
+  target_include_directories(hpx_includes INTERFACE ${CMAKE_SOURCE_DIR})
```

Comment:
@hkaiser note that I've cheated here! I don't understand why I need this (all the other modules are happy without it). It may be that this problem goes away by itself, but let me at least try once more to get rid of it before we merge this.

Reply:
I still don't get this. Let's go with this for now. It will definitely disappear once all the modules are in place.

Reply:
The reason is the `NOLIBS` option. I'm cleaning this up in #4147.